mmv1/third_party/terraform/services/netapp/resource_netapp_volume_test.go

```diff
@@ -0,0 +1,332 @@
+  storage_pool = google_netapp_storage_pool.default2.name
+  protocols = ["NFSV3"]
+  smb_settings = []
+  unix_permissions = "0740"
+  labels = {}
+  description = ""
+  snapshot_directory = true
+  security_style = "UNIX"
+  kerberos_enabled = false
+  export_policy {
+    rules {
+		access_type           = "READ_WRITE"
+		allowed_clients       = "0.0.0.0/0"
+		has_root_access       = "true"
+		kerberos5_read_only   = false
+		kerberos5_read_write  = false
+		kerberos5i_read_only  = false
+		kerberos5i_read_write = false
+		kerberos5p_read_only  = false
+		kerberos5p_read_write = false
+		nfsv3                 = true
+		nfsv4                 = false
+    }
+  }
+  restricted_actions = ["DELETE"]
+  snapshot_policy {
+    daily_schedule {
+      hour              = 1
+      minute            = 1
+      snapshots_to_keep = 1
+    }
+    enabled = true
+    hourly_schedule {
+      minute            = 10
+      snapshots_to_keep = 1
+    }
+    monthly_schedule {
+      days_of_month     = "1"
+      hour              = 1
+      minute            = 1
+      snapshots_to_keep = 1
+    }
+    weekly_schedule {
+      day               = "Sunday"
+      hour              = 1
+      minute            = 1
+      snapshots_to_keep = 1
+    }
+  }
+
+}
+
+data "google_compute_network" "default" {
+  name = "%{network_name}"
+}
+`, context)
+}
+
+func testAccNetappvolume_volumeCreateExample_updatesnapshot(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+
+resource "google_netapp_storage_pool" "default" {
+  name = "tf-test-test-pool%{random_suffix}"
+  location = "us-central1"
+  service_level = "PREMIUM"
+  capacity_gib = "2048"
+  network = data.google_compute_network.default.id
+}
+
+resource "google_netapp_storage_pool" "default2" {
+	name = "tf-test-pool%{random_suffix}"
+	location = "us-central1"
+	service_level = "EXTREME"
+	capacity_gib = "2048"
+	network = data.google_compute_network.default.id
+  }
+
+resource "google_netapp_volume" "test_volume" {
+  location = "us-central1"
+  name = "tf-test-test-volume%{random_suffix}"
+  capacity_gib = "200"
+  share_name = "testshare"
+  storage_pool = google_netapp_storage_pool.default2.name
+  protocols = ["NFSV3"]
+  smb_settings = []
+  unix_permissions = "0740"
+  labels = {}
+  description = ""
+  snapshot_directory = true
+  security_style = "UNIX"
+  kerberos_enabled = false
+  export_policy {
+    rules {
+		access_type           = "READ_WRITE"
+		allowed_clients       = "0.0.0.0/0"
+		has_root_access       = "true"
+		kerberos5_read_only   = false
+		kerberos5_read_write  = false
+		kerberos5i_read_only  = false
+		kerberos5i_read_write = false
+		kerberos5p_read_only  = false
+		kerberos5p_read_write = false
+		nfsv3                 = true
+		nfsv4                 = false
+    }
+  }
+  restricted_actions = ["DELETE"]
+  snapshot_policy {
+    daily_schedule {
+      hour              = 1
+      minute            = 1
+      snapshots_to_keep = 1
+    }
+    enabled = false
```

Comment:
I think we need to unset the other fields when `enabled` is `false`, based on the test failure reason.

Reply:
What is the error you are seeing? in my local tests, I can flip `enabled` on/off just fine. The logic behind it is too, that the schedule can stay in place as is you you just enabled/disable it. I cannot create an issue in my local setup by playing with enabled.

Reply:
I ran another test. I took the snapshot_policy block from the three tests and applied them (each one individual) on my existing volume without any issues. Whatever fails in the test, it doesn't look like anything related to snapshot_policy block.
Can you please share the error log for the failure? @rileykarson

Reply:
```
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # google_netapp_volume.test_volume will be updated in-place
          ~ resource "google_netapp_volume" "test_volume" {
                id                 = "projects/ci-test-project-188019/locations/us-central1/volumes/tf-test-test-volumenmell93m7h"
                name               = "tf-test-test-volumenmell93m7h"
                # (21 unchanged attributes hidden)
        
              + snapshot_policy {
                  + enabled = false
        
                  + daily_schedule {
                      + hour              = 1
                      + minute            = 1
                      + snapshots_to_keep = 1
                    }
        
                  + hourly_schedule {
                      + minute            = 10
                      + snapshots_to_keep = 1
                    }
        
                  + monthly_schedule {
                      + days_of_month     = "1"
                      + hour              = 1
                      + minute            = 1
                      + snapshots_to_keep = 1
                    }
        
                  + weekly_schedule {
                      + day               = "Sunday"
                      + hour              = 1
                      + minute            = 1
                      + snapshots_to_keep = 1
                    }
                }
        
                # (1 unchanged block hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
```

Reply:
looking into this, it seems setting snapshot_policy = false, completely omits the block from API response. 
Hence we can see '+' sign in front of enable = false as well.

Reply:
@rileykarson Thank you for providing the information. I am now able to recreate that problem. It is, as Naman said. If snapshot_policy.enabled is set to false, the API will omit the snapshot_policy.*_schedule blocks.

What's the right way to teach the provider to do that? My guess is setting `ignore_read` on these blocks.

@G-NamanGupta running the acceptance tests locally helps to uncover the error Riley reported too. We need to run the local acceptance tests before pushing ...

Reply:
Found https://googlecloudplatform.github.io/magic-modules/develop/field-reference/#ignore_read. 
need to confirm with @rileykarson for the usage though.

Reply:
I did test that already by simply giving ignore_read: rue to the (hourly|daily|weekly|monthy)_schedule blocks, but it didn't work in my environment. As mentioned in the link, we may have to try with a customer flatten function. The links to the example in the documentation is broken.

Reply:
We don't expect to enumerate rules like this in the provider- if a user sends a message to the API and a different one is returned back (that is not just a canonical form of the message, like upcasing or downcasing values), we want to show a diff. `ignore_read` is for fields that are _never_ returned. In the wild, a user would apply this config, get the diff, and the correct it.

Our tests don't allow diffs, so we just need to fix the test to specify a valid configuration.

Reply:
Nah, come on. That's too easy ;)

What you say makes perfect sense. We could have guessed that ourselves, but getting your thought is very helpful. Thank you!